EventsExpress/ClientApp/src/components/inventory/inventory.js

```diff
@@ -0,0 +1,125 @@
+import './inventory.css';
+
+const renderInventories = ({ fields, unitOfMeasuringState }) => {
+
+    return (
+    <div className="form-group">
+        <button type="button" title="Remove item" class="btn btn-secondary btn-icon" onClick={() => fields.push({})}>
+            <span class="icon"><i class="fas fa-plus"></i></span> Add item
+        </button>
+        <ul className="">
+        {fields.map((item, index) =>
+            <li className="" key={index}>
+                <div className="d-flex flex-wrap justify-content-between align-items-center">
+                    <div className="p-2 bd-highlight align-self-end">
+                        <span>{index + 1}.</span>
+                    </div>
+                    <div className="p-2 bd-highlight">
+                        <Field
+                            name={`${item}.itemName`}
+                            type="text"
+                            fullWidth={false}
+                            label="Item name"
+                            component={renderTextField}/>
+                    </div>
+                    <div className="p-2 bd-highlight">
+                        <Field
+                            name={`${item}.needQuantity`}
+                            type="number"
+                            fullWidth={false}
+                            label="count"
+                            component={renderTextField}/>
+                    </div>
+                    <div className="p-2 bd-highlight">
+                        <Field
+                            className="selectpicker"
+                            name={`${item}.unitOfMeasuring.id`} 
+                            component={renderSelectField}>
+                            <option></option>
+                            {unitOfMeasuringState.units.map((unit, key) => 
+                                <option value={unit.id} key={key}>{unit.unitName}</option>
+                            )} 
+                        </Field>
+                    </div>
+                    <button type="button" title="Remove item" class="p-2 btn btn-circle clear-backgroud align-self-end" onClick={() => fields.remove(index)}>
+                        <i class="fas fa-trash red"></i>
+                    </button>
+                </div>
+            </li>
+        )}
+        </ul>
+    </div>
+    )
+}
+
+
+
+class Inventory extends Component {
+
+    constructor() {
+        super();
+
+        this.state = {
+            isOpen: true
+        };
+
+        this.handleOnClickCaret = this.handleOnClickCaret.bind(this);
+    }
+
+    componentWillMount() {
+        this.props.get_unitsOfMeasuring();
+    }
+
+    handleOnClickCaret() {
+        this.setState(state => ({
+            isOpen: !state.isOpen
+        }));
+    }
+
+    render() {
+        return (
+            <div>
+                <div className='d-flex justify-content-start align-items-center'>
```

Comment:
This code is duplicated between display and edit. Consider extracting it to a separate component.

Reply:
Solved